spark-streaming-wordcount-java-processor/README.md

```diff
@@ -0,0 +1,62 @@
+Spring XD spark streaming processor example
+=================
+
+### Setup and configure spark cluster
+
+Spark streaming module can be run on any of the supporting cluster modes.
+You can either setup spark `standalone` cluster or use `mesos`, `yarn`. For instructions, see [spark cluster overview] (https://spark.apache.org/docs/1.2.1/cluster-overview.html) and [spark standalone setup] (https://spark.apache.org/docs/1.2.1/spark-standalone.html).
+
+For the development experience, you can use `local` mode as well.
+
+Once the spark cluster is setup, you can configure the property `spark.master` in XD servers.yml to use the corresponding cluster manager URL.The `spark.master` URL property can be configured at the module implementation using `@SparkConfig` annotation that returns Spark configuration properties:
+
+```
+  @SparkConfig
+	public Properties getSparkConfigProperties() {
+		Properties props = new Properties();
+		// Any specific Spark configuration properties would go here.
+		// These properties always get the highest precedence
+		props.setProperty(SPARK_MASTER_URL_PROP, "local[4]");
+		return props;
+	}
+```
+
+By default it is set to use `spark://localhost:7077`
+
+### Upload the module
+
+Spring XD provides uploading modules archive from the `shell` interface. We can use this approach to upload the word count example module into XD's module registry.
+
+1. Run the build to generate the jar
+
+  ```
+    ./gradlew clean build
+  ```
+  This will generate the spark-streaming-wordcount-java-processor-0.1.0.jar under build/libs.
+  
+2. Upload the generated jar into XD module registry
+
+  ```
+    module upload --file <location of the jar file>/spark-streaming-wordcount-java-processor-0.1.0.jar 
```

Comment:
Both for Scala and Java: We should use a path that starts at the project directory the project and includes the `build/libs` folder explicitly. It's likely that some users won't know where to look for the JAR. Will fix in merge.